### DISCUSSION TITLE
Support for Preload in Image component

### DISCUSSION THREAD

**Discussion by bitforcesrl** [Shopify/hydrogen-ui#105] [Ideas + Feature Requests]
@frehner The image component is aware of all infos needed to add a preloader tag in the header to preload the image [image preloading](https://web.dev/preload-responsive-images/)

It would be great to add an input param <Image preload="true" ...props/> to push in the head the needed script.
It's possible to preload also responsive images with sourceset like the code wil look like this
```
<link rel=preload as=image fetchPriority="high"
      imagesizes="(min-width: 96em)  33vw, 45vw"
      imagesrcset="https://cdn.shopify.com/s/files/1/0551/5290/2227/articles/Banner_fw22_blog_800x_d79e2ca4-2fef-4a5e-91ca-e709e8fde032.webp?v=1672237102&width=80&height=53.33333333333333&crop=center 80w, ....">
 ```
      
`imagesizes` comes from Image's sizes prop and `imagesrcset` comes from the internal Image function that generate all cdn.shopify image URLs using this Image's widths prop
      
we can now leverage the new flag `fetchPriority="high"` which will imrprove more the LCP score

**Comment by frehner** [comment]
Hydrogen-UI is a toolset that works across React-based frameworks, like Next, Remix, etc. So this would be difficult to do here, as we would have to build an adapter for each framework. At the moment, we don't plan on doing things like that, so this sort of functionality would probably have to live in your code for now.
